logstash-core/src/test/java/org/logstash/secret/store/SecretStoreFactoryTest.java

```diff
@@ -58,46 +58,45 @@ private static void setEnv(Map<String, String> newenv) throws Exception {
     }
 
     @Test
-    public void testAlternativeImplementation(){
-        String beforeSystem = System.getProperty(SecretStore.class.getCanonicalName());
-        try {
-            System.setProperty(SecretStore.class.getCanonicalName(), "org.logstash.secret.store.SecretStoreFactoryTest$MemoryStore");
-            SecretStore secretStore = SecretStoreFactory.loadSecretStore(new SecureConfig());
-            assertThat(secretStore).isInstanceOf(MemoryStore.class);
-            validateMarker(secretStore);
-        } finally {
-            if (beforeSystem == null) {
-                System.clearProperty(SecretStore.class.getCanonicalName());
-            } else {
-                System.setProperty(SecretStore.class.getCanonicalName(), beforeSystem);
-            }
-        }
+    public void testAlternativeImplementation() {
+        SecureConfig secureConfig = new SecureConfig();
+        secureConfig.add("keystore.classname", "org.logstash.secret.store.SecretStoreFactoryTest$MemoryStore".toCharArray());
```

Comment:
Maybe style change, but the test would fail so maybe it's ok, but saying `SecretStoreFactoryTest.MemoryStore.getName().toCharArray()` instead of a hardcoded string -- that way this may fail at compile-time instead of run-time?

(this is an optional correction to make, assuming you agree with the suggestion)

Reply:
agreed, fixed with next review.